### PR TITLE
[CCR] Clear fetch exceptions if an empty but successful shard changes response returns

### PR DESCRIPTION
**Pull request by martijnvg**
Also fixed ShardFollowNodeTaskTests to not return ops when responseSize
is empty. Otherwise ops are returned when no ops are expected to be returned.

### CODE REVIEW

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTaskTests.java [hunk 1]
```diff
@@ -228,6 +229,35 @@ public void testReceiveRetryableError() {
         assertThat(status.leaderGlobalCheckpoint(), equalTo(63L));
     }
 
+    public void testEmptyShardChangesResponseShouldClearFetchException() {
```
Comment (bleskes) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTaskTests.java]: I think we generilize this and test that any response (empty or not) clears errors.

**Review by bleskes — APPROVED** [review]
LGTM. Left a suggestion. Also note that the build has failed.

**Review by jasontedor — APPROVED** [review]
LGTM.

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTaskTests.java (hunk 1 above)
Comment (martijnvg) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTaskTests.java]: This is tested in `testReceiveRetryableError()`.

**Pull request merged**